changelog/5275.feature.rst

```diff
@@ -0,0 +1 @@
+Allows :meth:`sunpy.map.GenericMap.draw_quadrangle` to accept :ref:`u.pix <astropy:astropy-units>` as input to enable plotting boxes in the pixel space of both the native map and other maps when the correct transform is specified.
```

Comment:
```suggestion
Allows :meth:`sunpy.map.GenericMap.draw_quadrangle` to accept pixel units as input to enable plotting boxes in the pixel space of the map, which can be different from the plot axes.
```
With the changes below, the "correct transform" is automatic.  Also, forget an intersphinx link.  I can't fine anything clean to link.